examples/ui/myapp_component.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+)
+
+// Config holds configuration used by MyAppComponent.
+type Config struct {
+	BindAddr                string
+	SiteDomain              string
+	GracefulShutdownTimeout time.Duration
+}
+
+// NewConfig initialises hard-coded configuration for MyAppComponent.
+func NewConfig() *Config {
+	return &Config{
+		BindAddr:                ":26601",
+		SiteDomain:              "localhost",
+		GracefulShutdownTimeout: 5 * time.Second,
+	}
+}
+
+// MyAppComponent holds the initialized http server and config required for running component tests.
+type MyAppComponent struct {
+	errorChan      chan error
+	Config         *Config
+	HTTPServer     *http.Server
+	ServiceRunning bool
+}
+
+// NewMyAppComponent initializes server used for running component.
+func NewMyAppComponent(handler http.Handler) *MyAppComponent {
+	ctx := context.Background()
+
+	c := &MyAppComponent{
+		errorChan: make(chan error, 1),
+		Config:    NewConfig(),
+		HTTPServer: &http.Server{
+			Handler: handler,
+		},
+		ServiceRunning: false,
+	}
+
+	c.run(ctx)
+	c.ServiceRunning = true
+
+	return c
+}
+
+func (c *MyAppComponent) run(ctx context.Context) {
```

Comment:
Is contexct supposed to be used, or is the argument not needed?

Reply:
Removed